temporalio/ui
# DT-1016 - display custom search attributes in UI

## Description & motivation 💭 
Previously the UI was using `workflowservice/GetSearchAttributes` to fetch Search Attributes at the cluster level. By adding and consuming the `operatorservice/ListSearchAttributes` API, we are able to fetch Search Attributes on a per-namespace level. This PR has no style or design changes, just rewires our service and store(s) to use the new RPC method and does some more test cleanup.

Other changes of note:
- update `@temporalio/api` submodule to `v1.21.0`
- update other ui-server go dependencies
- update `tests/integration` to use only mocks
### Screenshots (if applicable) 📸 
N/a
### Design Considerations 🎨 
N/a
## Testing 🧪 

### How was this tested 👻 
Covered by manual testing and existing e2e/unit tests. No new tests were added.

### Steps for others to test: 🚶🏽‍♂️🚶🏽‍♀️ 
- go to `/namespaces/{namespace}` and ensure all system search attributes are displayed
- add a search attribute (can be done with `temporal operator search-attribute create --namespace {namespace} --name SearchAttrName --type SearchAttrType` if running against temporal cli's dev server)
- refresh `/namespaces/{namespace}` to ensure new custom search attribute is displayed
- also go to `/namespaces/{namespace}/workflows` and open the workflows summary table configuration panel to ensure all search attributes are displayed.
## Checklists

### Draft Checklist 
N/a
### Merge Checklist 
N/a
### Issue(s) closed 
closes #1274
## Docs

### Any docs updates needed?
